### PR TITLE
Feature/usernames conversation menu

### PR DESCRIPTION
**Pull request by daehn**
# What's in this PR?

* This splits `UserConnectionView` in two separate classes, `UserConnectionView` and `IncomingConnectionView`. While `UserConnectionView` does not contain any controls anymore (the cancel request and archive button have been moved into a separate class called `OutgoingConnectionBottomBar`) and is shown at the top of 1-on-1 conversations, the `IncomingConnectionView` is used to show incoming requests and has a ignore and accept button. The controls for outgoing connection requests have been moved into a separate control to make the header more reusable and fix layout issues for outgoing requests.
* Add functionality to use an arbitrary UIView as title view in an `ActionSheetController`, this was needed in order to use a `UserNameDetailView` (containing the name, handle and correlation labels) at the top instead of a simple label.
* Copy adjustments (invitations & search).

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/Components/Controllers/ActionSheetController/ActionSheetController.m [hunk 1]
```diff
@@ -113,6 +113,7 @@ @interface ActionSheetController ()
 @property (nonatomic, readonly) ActionSheetControllerLayout layout;
 @property (nonatomic, readonly) ActionSheetContainerView *actionSheetContainerView;
 @property (nonatomic, readonly) UIView *sheetView;
+@property (nonatomic) UIView *titleView;
```
Comment (typfel) [inline Wire-iOS/Sources/UserInterface/Components/Controllers/ActionSheetController/ActionSheetController.m]: Can't be `readonly`?

File: Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController.m [hunk 2]
```diff
@@ -248,23 +248,10 @@ - (void)updateTableViewHeaderView
...
-                    break;
-                default:
-                    [self updateTableViewHeaderView];
-                    break;
-            }
-        };
+    BOOL connectionOrOneOnOne = self.conversation.conversationType == ZMConversationTypeConnection || self.conversation.conversationType == ZMConversationTypeOneOnOne;
 
+    if (connectionOrOneOnOne && nil != otherParticipant) {
+        self.connectionViewController = [[UserConnectionViewController alloc] initWithUserSession:[ZMUserSession sharedSession] user:otherParticipant];
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController.m]: So the buttons to cancel connection / block are gone from this view now?

Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/ConversationContentViewController.m]: @daehn I see it's moved to an overlay in `ConversationViewController` now.

File: Wire-iOS/Sources/UserInterface/Conversation/ConversationViewController.m [hunk 3]
```diff
@@ -235,6 +238,55 @@ - (void)createContentViewController
...
+}
+
+- (void)updateOutgoingConnectionVisibility
+{
+    if (nil == self.conversation) {
+        return;
+    }
+
+    BOOL outgoingConnection = self.conversation.relatedConnectionState == ZMConnectionStatusSent;
+    self.contentViewController.tableView.scrollEnabled = !outgoingConnection;
```
Comment (mikeger) [inline Wire-iOS/Sources/UserInterface/Conversation/ConversationViewController.m]: Funny, I asked Yorgos and he said last time that it is fine that it is remaining scrollable as conversation content.

Comment (daehn) [inline Wire-iOS/Sources/UserInterface/Conversation/ConversationViewController.m]: Yeah, he changed his mind ;)

**Comment by daehn** [comment]
@typfel @mikeger please re-review.

**Pull request merged**